### PR TITLE
Add keyed NT Data and listeners

### PR DESCRIPTION
**Pull request by Bankst**

### CODE REVIEW

**Comment by Bankst** [comment]
[Fill in NTDataConsumer](https://trello.com/c/hMHp2kT6/49-fill-in-ntdataconsumer)

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataChangeListener.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.photonvision.common.dataflow.networktables;
+
+public class NTDataProvider {
+
+}
```
Comment (mcm001) [inline photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataChangeListener.java:5]: What's this class for?

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        targetPitchEntry.forceSetDouble(bestTarget.getPitch());
+        targetYawEntry.forceSetDouble(bestTarget.getYaw());
+        targetAreaEntry.forceSetDouble(bestTarget.getArea());
+
+        var poseX = bestTarget.getRobotRelativePose().getTranslation().getX();
+        var poseY = bestTarget.getRobotRelativePose().getTranslation().getY();
+        var poseRot = bestTarget.getRobotRelativePose().getRotation().getDegrees();
+        targetPoseEntry.forceSetDoubleArray(new double[] { poseX, poseY, poseRot });
+    }
```
Comment (mcm001) [inline photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java:130]: I would flush() only at the end -- all the bytepacked stuff will be sent together anyways, and we're rate limited to 100hz

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataChangeListener.java (hunk 1 above)
Comment (Bankst) [reply photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataChangeListener.java:5]: That must have slipped in. It's getting renamed `NTDataChangeListener` and will house the listeners for `pipelineIndex`, `driverMode`, etc

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java (hunk 2 above)
Comment (Bankst) [reply photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java:130]: Good point, I will change that.

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            // TODO: Log
+            return;
+        }
+
+        driverModeConsumer.accept(newDriverMode);
+        // TODO: Log
+    }
+
+    private void updateEntries() {
+        rawDataEntry = subTable.getEntry("rawData");
```
Comment (prateekma) [inline photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java:103]: The name of the entry should be `rawBytes` (this is how it is in photonlib).

Comment (Bankst) [reply photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java]: Resolved in 71f99d5

File: photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java (hunk 2 above)
Comment (Bankst) [reply photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java:163]: Resolved in 71f99d5

**Pull request merged**